nucypher/network/nodes.py

```diff
@@ -911,7 +912,7 @@ def __init__(self,
         self.__worker_address = None
 
         if substantiate_immediately:
-            self.substantiate_stamp(client_password=password)
+            self.substantiate_stamp(client_password=password, checksum_address=worker_address)
```

Comment:
:+1:

Reply:
Feels a bit odd to specify the `checksum_address` of the account you're wanting to sign with here to substantiate a stamp. Shouldn't this be handled at a lower level?

Reply:
Yeah, @tuxxy is right here. This should be handled at a lower level, but the problem is that `BlockchainPower` seems to be client-specific now, not account-specific, so at this point the Teacher doesn't know anymore which account corresponds her `BlockchainPower`.

Reply:
The problem I was having here is that worker address is only accessible from the local-scoped variable `worker_address`, since it has not yet been set on `self`.

Reply:
Resolved in 11e79d0 :heavy_check_mark: